woocommerce/woocommerce
# Consistent folder structure for E2E and API test results

### All Submissions:


### Changes proposed in this Pull Request:

Closes #35914.

Currently, the reports generated by our E2E and API tests are located in several different places. Our E2E test suite saves the test outputs in these places:

<img width="185" alt="wPTuOwONq0" src="https://user-images.githubusercontent.com/4509348/206822659-49dcf2fa-94f5-4eba-a7ee-89e5581e183a.png">

While the API tests save the outputs to these folders:

<img width="243" alt="g53h9j08AE" src="https://user-images.githubusercontent.com/4509348/206822669-ae433085-8eaa-4bc1-a6d5-dade0c83b780.png">

This PR would save these outputs in a single `test-results` directory, like so:

<img width="193" alt="YLCYpyWGUF" src="https://user-images.githubusercontent.com/4509348/206822650-355b2bad-1999-43a3-a535-659a3a30936c.png">

### How to test the changes in this Pull Request:

1. Clone this branch and launch a local E2E environment using the setup instructions in the [README](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/README.md#introduction).
2. To test the E2E & API output directories, navigate to `plugins/woocommerce` and run:
    ```bash
    pnpm exec playwright test --config=tests/e2e-pw/daily.playwright.config.js basic.spec.js
    pnpm run test:api-pw -- hello.test.js
    ```
    We don't need to run the entire E2E and API test suites because we're only concerned with where the Playwright and Allure test reports are going to be saved.
4. After they're run, verify that the E2E and API test reports were saved in their respective `test-results` folders like so:
    <img width="193" alt="YLCYpyWGUF" src="https://user-images.githubusercontent.com/4509348/206823960-93544ab2-bc5a-4589-bdb2-205b2cd7edbe.png">
    <img width="121" alt="pAB9Y6IMWz" src="https://user-images.githubusercontent.com/4509348/206823965-c5ee86c6-5542-48b4-b62a-758588c69351.png">
    Note that the difference between the `test-results/playwright-report` and `test-results/report` folders is that `playwright-report` contains the static HTML files generated by Playwright, while the `report` folder contains the data for generating these HTML files.
1. Go to the "Checks" tab of this PR, and re-run the "Run tests against PR" check.
1. Wait for the workflow to finish, and wait for 1-2 minutes to complete the publishing of the PR's E2E and API Allure reports.
2. Navigate to the Allure report pages and make sure that their title shows the correct date and time you ran the workflow:
    - E2E: https://a8c-woo-test-reports.s3.amazonaws.com/public/pr/35907/e2e/index.html
    - API: https://a8c-woo-test-reports.s3.amazonaws.com/public/pr/35907/api/index.html
1. Navigate to https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-daily.yml?query=branch%3Ae2e%2Foutput-directories and run the "Smoke test daily" workflow using this branch (`e2e/output-directories`)
    <img width="974" alt="Qd9KgdW9Bq" src="https://user-images.githubusercontent.com/4509348/206824448-df4f3ce0-b39e-439b-9e05-72efbe9d4bdf.png">
1. Wait for the workflow to finish.
2. Wait for 1-2 minutes to complete the publishing of the E2E and API Allure reports.
3. Go to their respective E2E and API Allure reports and make sure that their title shows the correct date and time when you triggered the daily smoke test workflow:
    - E2E: https://a8c-woo-test-reports.s3.amazonaws.com/public/daily/e2e/index.html
    - API: https://a8c-woo-test-reports.s3.amazonaws.com/public/daily/api/index.html
    <img width="884" alt="APJa8OZspy" src="https://user-images.githubusercontent.com/4509348/206824574-51f9b694-418c-4e88-b3ee-cbf3f4810240.png">
1. Check for mistakes in these 2 README's:
    - plugins/woocommerce/tests/api-core-tests/README.md
    - plugins/woocommerce/tests/e2e-pw/README.md


### Other information:




### FOR PR REVIEWER ONLY:
